src/libawkward/array/IndexedArray.cpp

```diff
@@ -1980,6 +1980,216 @@ namespace awkward {
+                                         const Index64& starts,
+                                         const Index64& parents,
+                                         int64_t outlength,
+                                         bool ascending,
+                                         bool stable,
+                                         bool keepdims) const {
+    int64_t numnull;
+    struct Error err1 = util::awkward_indexedarray_numnull<T>(
+      &numnull,
+      index_.ptr().get(),
+      index_.offset(),
+      index_.length());
+    util::handle_error(err1, classname(), identities_.get());
+
+    Index64 nextparents(index_.length() - numnull);
+    Index64 nextcarry(index_.length() - numnull);
+    Index64 outindex(index_.length());
+    struct Error err2 = util::awkward_indexedarray_reduce_next_64<T>(
+      nextcarry.ptr().get(),
+      nextparents.ptr().get(),
+      outindex.ptr().get(),
+      index_.ptr().get(),
+      index_.offset(),
+      parents.ptr().get(),
+      parents.offset(),
+      index_.length());
+    util::handle_error(err2, classname(), identities_.get());
+
+    ContentPtr next = content_.get()->carry(nextcarry);
+    ContentPtr out = next.get()->sort_next(negaxis,
+                                           starts,
+                                           nextparents,
+                                           outlength,
+                                           ascending,
+                                           stable,
+                                           keepdims);
+
+    Index64 nextoutindex(index_.length());
+    int64_t j = 0;
+    for (int64_t i = 0; i < parents.length(); i++) {
+      int64_t parent = parents.getitem_at_nowrap(i);
+      int64_t start = starts.getitem_at_nowrap(parent);
+      int64_t nextparent = nextparents.getitem_at_nowrap(j);
+      if (parent == nextparent) {
+        nextoutindex.setitem_at_nowrap(i, j);
+        ++j;
+      }
+      else {
+        nextoutindex.setitem_at_nowrap(i, -1);
+      }
+    }
+
+    out = std::make_shared<IndexedOptionArray64>(Identities::none(),
+                                                 util::Parameters(),
+                                                 nextoutindex,
+                                                 out);
+
+    std::pair<bool, int64_t> branchdepth = branch_depth();
+    if (!branchdepth.first  &&  negaxis == branchdepth.second) {
+      return out;
+    }
+    else {
+      if (RegularArray* raw =
+        dynamic_cast<RegularArray*>(out.get())) {
+        out = raw->toListOffsetArray64(true);
+      }
+      if (ListOffsetArray64* raw =
+        dynamic_cast<ListOffsetArray64*>(out.get())) {
+        Index64 outoffsets(starts.length() + 1);
+        if (starts.length() > 0  &&  starts.getitem_at_nowrap(0) != 0) {
+          throw std::runtime_error(
+            "sort_next with unbranching depth > negaxis expects a "
+            "ListOffsetArray64 whose offsets start at zero");
+        }
```

Comment:
This use of `getitem_at_nowrap` is allowed because it only gets one value (the first). If the data were on a GPU, a kernel would be launched to get that individual value, but 10 μs is not a huge overhead to add at the beginning of "sort this whole array for me."

Reply:
agree, thanks!